clients/cobol-lsp-vscode-extension/src/commands/RenumCommand.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2025 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+import * as vscode from "vscode";
+
+export interface RenumberParameters {
+  start: number;
+  end: number;
+  digits: number;
+}
+
+const maxLines = 999999;
+
+export const RENUM_LEFT: RenumberParameters = {
+  start: 0,
+  end: 6,
+  digits: 6,
+};
+export const RENUM_RIGHT: RenumberParameters = {
+  start: 72,
+  end: 80,
+  digits: 8,
+};
+
+export type RenumDocument = {
+  lineCount: number;
+  lineAt: (n: number) => {
+    text: string;
+  };
+};
+
+/**
+ * Renumber 1-7 or 73-80 columns in active editor.
+ *
+ * @param document RenumDocument
+ * @param edit  vscode.TextEditorEdit
+ * @param params  RenumberParameters
+ */
+export function renumberLines(
+  document: RenumDocument,
+  edit: vscode.TextEditorEdit,
+  params: RenumberParameters,
+) {
+  const lineCount = document.lineCount;
+  if (lineCount > maxLines) return;
```

Comment:
If this ever happens we should issue an error message to let the user know about the limit.